zkevm-circuits/src/evm_circuit.rs

```diff
@@ -224,7 +224,22 @@ pub mod test {
                         .assign(&mut region, offset, &Default::default())?;
                     offset += 1;
 
-                    for rw in rws.0.values().flat_map(|rws| rws.iter()) {
+                    let mut rows = rws
+                        .0
+                        .values()
+                        .flat_map(|rws| rws.iter())
+                        .collect::<Vec<_>>();
+
+                    rows.sort_by_key(|a| a.get_rw_counter());
+                    let mut pre_rw_counter = 0;
+                    for rw in rows {
+                        if pre_rw_counter == 0 {
+                            assert!(rw.get_rw_counter() == 1);
+                        } else {
+                            assert!(rw.get_rw_counter() == pre_rw_counter + 1);
+                        }
+
+                        pre_rw_counter = rw.get_rw_counter();
```

Comment:
```suggestion
                    let mut expected_rw_counter = 1;
                    for rw in rows {
                        assert!(rw.get_rw_counter() == expected_rw_counter);
                        expected_rw_counter += 1;
```
I believe your implementation is correct, I'm just suggesting a simplification. Feel free to use it or ignore it :)

Reply:
nice to me,  apply it in [2d483e7](https://github.com/appliedzkp/zkevm-circuits/pull/415/commits/2d483e7dc2f5c95d7c911b2243afc4fb6fe8a0d2) thanks